generators/client/templates/vue/src/main/webapp/app/main.ts.ejs

```diff
@@ -69,6 +69,14 @@ const gatewayService = new GatewayService();
 <%_ } %>
 const principal = new Principal(store, <%_ if (enableTranslation) { _%>translationService, i18n, <%_ } _%><%_ if (authenticationType === 'session') { _%>(<any>Vue).cookie, <%_ } _%><%_ if (websocket === 'spring-websocket') { _%>trackerService, <%_ } _%>router);
 
+router.beforeEach((to, from, next) => {
```

Comment:
@jhipster/vuejs-developers Would it be better to put that into `app/router/index.ts` to have all router related stuff in one file? For the upcoming access check we need e.g. the principal which also needs the router...